implementation-details.md

```diff
@@ -424,6 +426,8 @@ When a node receives a pull response it processes its list of `CrdsValue`s:
 
 Nodes ping their peers from time to time to see whether they are active. They create a ping message which contains a randomly generated 32-byte token. The peer receiving the ping should respond with a pong message that contains a hash of the received ping token within a certain amount of time. If the peer fails to respond, it will not receive any other message from the node who pinged it. Otherwise, the origin of the ping message will store the received pong in a cache.
 
+### Gossip loop
+The gossip loop is an infinite loop, run by a node in a separate thread. The node generates there push messages, pull requests and pings, and refreshes its active set of nodes.
```

Comment:
nit: I'd change this to:
The gossip loop is an infinite loop executed by a node in a separate thread. In this loop, the node generates push messages, pull requests, and pings, while also refreshing its active set of nodes.